### PR TITLE
Fix: #129 add option to search PDF

### PR DESCRIPTION
**Pull request by ac491**
# Description
#129 Added search bar to search for PDF files

screenshot_2018-06-30-16-25-03-798_swati4star createpdf 1

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ x ] My changes generate no new warnings

### CODE REVIEW

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import android.app.Activity;
 import android.content.Context;
 import android.content.DialogInterface;
+import android.nfc.Tag;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: What is the need of this import?

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java [hunk 2]
```diff
@@ -109,10 +112,45 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
     public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         super.onCreateOptionsMenu(menu, inflater);
         inflater.inflate(R.menu.activity_view_files_actions, menu);
+        MenuItem item = menu.findItem(R.id.action_search);
+        mSearchView = (SearchView) item.getActionView();
+        mSearchView.setQueryHint("Name of file");
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: Add `"Name of file"` to `strings.xml` and use that here

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java [hunk 3]
```diff
@@ -109,10 +112,45 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
...
+        mSearchView.setSubmitButtonEnabled(true);
+        mSearchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String s) {
+                ArrayList searchResult = mFileUtils.searchPDF(s);
+                if(searchResult.isEmpty()) {
+                    Toast.makeText(mActivity,"Couldn't find anything",Toast.LENGTH_LONG).show();
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: Move the constant string `"Couldn't find anything"`  to `strings.xml`

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 4]
```diff
@@ -263,5 +265,45 @@ public void openFile(File file) {
         }
     }
 
+    /**
+     * Used in searchPDF to give the closest result to search query
+     * @param query - Query from search bar
+     * @param fileName - name of PDF file
+     */
+    private int checkChar(String query,String fileName) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: add space after `,`

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 5]
```diff
@@ -263,5 +265,45 @@ public void openFile(File file) {
         }
     }
 
+    /**
+     * Used in searchPDF to give the closest result to search query
+     * @param query - Query from search bar
+     * @param fileName - name of PDF file
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: add @return also in the comment

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 6]
```diff
@@ -263,5 +265,45 @@ public void openFile(File file) {
...
+        if( q.containsAll(f) || f.containsAll(q) ) {
+            return 1;
+        }
+        return 0;
+    }
+
+    /**
+     * Used to search for PDF matching the search query
+     * @param query - Query from search bar
+     */
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: Add comment for @return also here

File: app/src/main/res/menu/activity_view_files_actions.xml [hunk 7]
```diff
@@ -2,16 +2,27 @@
 <menu xmlns:android="http://schemas.android.com/apk/res/android"
       xmlns:app="http://schemas.android.com/apk/res-auto">
 
+    <item
+        android:id="@+id/action_search"
+        android:icon="@drawable/search"
+        android:orderInCategory="1"
+        android:title="search"
```
Comment (Swati4star) [inline app/src/main/res/menu/activity_view_files_actions.xml]: Move `search` to `strings.xml`

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java (hunk 1 above)
Comment (ac491) [reply app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: Oh sorry, I'll remove it. It was to log errors

**Comment by ac491** [comment]
should I remove the .pdf part too, from file name when searching for the query?

**Comment by Swati4star** [comment]
Yes. @ac491

**Comment by Swati4star** [comment]
Merged in 7238c7c8499915dbc7bcdbcc9bb4bd83f88d25bb 👍

**Pull request closed without merging**